src/EntityFramework6.Npgsql/SqlGenerators/VisitedExpression.cs

```diff
@@ -526,7 +526,7 @@ internal override void WriteSql(StringBuilder sqlText)
             }
             else
             {
-                var wrap = !(_from is LiteralExpression || _from is ScanExpression);
+                var wrap = !(_from is LiteralExpression || _from is ScanExpression || _from is FunctionExpression);
```

Comment:
Is avoiding the wrapping by parens necessary ? I'm not familiar with the pgsql for calling TVF sprocs so just making sure.